### ISSUE TITLE
pull --insecure still verifies cert

### ISSUE THREAD

**Issue by ScreamingDev** [git-ftp/git-ftp#366]
### Bug

* How to reproduce the issue?

Place an invalid cert somewhere for FTP download.

### Environment

Please copy the output of the following commands.

```sh
$ echo "$SHELL"
/bin/bash
$ uname -a
Linux rmp 4.4.0-81-generic #104-Ubuntu SMP Wed Jun 14 08:17:06 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
$ git-ftp version
git-ftp version 1.4.0
$ lftp -v
LFTP | Version 4.6.3a | Copyright (c) 1996-2014 Alexander V. Lukyanov
```

The error is:

```
Tue Jul  4 14:04:33 CEST 2017: cd: Fatal error: Certificate verification: certificate common name doesn't match requested host name 'example.org'
mirror: Fatal error: Certificate verification: certificate common name doesn't match requested host name 'example.org'
1 error detected
```

but I run it with: `git ftp pull --insecure -v -P`

So why is this failing? 
It shall not check the cert or just accept it.

**Comment by mkllnk** [comment]
Yes, you are right. The pull action ignores the `--insecure` flag. It was simply forgotten. As a workaround, you can put `set ssl:verify-certificate false` in your `~/.lftprc` file. But that will deactivate the checking for all lftp commands unless you activate it explicitly when using lftp yourself.

**Comment by ScreamingDev** [comment]
Nice. With that in mind mirroring works again. Thanks!
I wonder if .gitignore will be used for mirroring but that's another thing I figure out now ;)

**Comment by marsjaninzmarsa** [comment]
Funny thing - issue exists even when specifying pure `ftp://` as protocol in config… so which cert they trying to verify? :sweat_smile:

**Comment by DerTanzschuh** [comment]
Looks like there is the same problem with `git ftp download`. I have tried it also with `ftp` and the error occurs.

**Comment by DerTanzschuh** [comment]
There are some more funny things going on here.

The reason why there is certificate to check even with `ftp`might be the handshake at the beginning of the FTP-Connection. Here the client and server negotiate features to be used. If both the client and the server support TLS, the server will send a certificate and the client might validate it. The debug log looks something like this:
```
---- Connecting to example.org (109.123.123.123) port 21
<--- 220 FTP Server ready.                         
---> FEAT
<--- 211-Features:                                       
<---  AUTH TLS
<---  CLNT
...
<---  UTF8
<--- 211 End
---> AUTH TLS
<--- 234 AUTH TLS successful                            
---> CLNT lftp/4.8.4
```
TLS support can be disabled on the client  by setting `ftp:ssl-allow false` as described above, but I thinks it’s a quite good feature. 

---

I have testes some things on my test server. The current command being set looks like this (I have shortened some parts but everything important is included):
```
lftp -d 
    -e "set ftp:list-options -a &&
        mirror -v --delete --exclude=... . ./ && 
        wait all && exit" 
    -u "user,paswd" "ftp://example.org/html/dev/ftp-test/"
```
There are two things I would like to point out:
* `$SYNCROOT* (in my case `html/dev/ftp-test`) is appended to the host
* source for the `mirror` command is specified as `.`

If I call the command like it is on a host with a valid certificate everything is fine and the content of `ftp-test` is downloaded as expected.
If I call this command on a server with a invalid certificate (as described above) the verification fails. OK. fine. We expected this.
If I set `ssl:verify-certificate false` in `~/.lftprc` it works as well. So far, so good.

**But I do not want to set this globally, but only for this one call.** That is theoretically possible, too. You can set this setting as well as the other one:
```
lftp -d 
    -e "set ssl:verify-certificate false && set ftp:list-options -a &&
        mirror -v --delete --exclude=... . ./ && 
        wait all && exit" 
    -u "user,paswd" "ftp://example.org/html/dev/ftp-test/"
```

**And now to the really funny stuff.**

With the valid certificate everything is still ok. But if I call it to the host with the invalid certificate suddenly **the content of the root folder is downloaded** (html and its neighbors). For information only: In fact it’s literally exactly the same server, just called with another domain.)

**What about the solution?**

Don’t include `$SYNCROOT` in the URL, but `cd` after connecting.
```
lftp -d 
    -e "set ssl:verify-certificate false && set ftp:list-options -a &&
        cd html/dev/ftp-test/ &&
        mirror -v --delete --exclude=... . ./ && 
        wait all && exit" 
    -u "user,paswd" "ftp://example.org/"
```
I have only tested this on one particular case so far. I think it might work with absolute paths as well. But it requires that the `$SYNCROOT` is configured properly. Some people include this in their host url configuration. 
If this is a limitation that we can accept, it would be a possible solution to use `--insecure`with lftp (`pull`, `download`, …).

**Comment by DerTanzschuh** [comment]
>But it requires that the `$REMOTE_ROOT` is configured properly. Some people include this in their host url configuration.
If this is a limitation that we can accept …

During the implementation I noticed that the url is actually split up in three parts: `$REMOTE_PROTOCOL`, `$REMOTE_HOST`, `$REMOTE_PATH`. **Therefore, this limitation is actually not there.**

**_Note:_** If the path is contained in the URL, `$REMOTE_PATH` is not set in `set_remote_path()`, as you might expect, but as a side effect of `set_remote_protocol()`. This seems a bit odd to me. `set_remote_path()` takes care of processing `$REMOTE_ROOT` and adding the trailing slash to `$REMOTE_PATH`. Maybe that's a good point for refactoring. But that's another issue. 

I just wanted to say here that the implementation from the commit also works if the path is specified in the URL. Oh. An absolute paths work, as well.

**Comment by DerTanzschuh** [comment]
This is fixed by #526
